hoomd/md/nlist.py

```diff
@@ -174,251 +158,127 @@ def __init__(self, buffer=0.4, exclusions=('bond',), rebuild_check_delay=1,
         super()._attach()
 
-    def _detach(self):
-        del self._cpp_cell
-        super()._detach()
 
-
-class stencil(nlist):
-    R""" Cell list based neighbor list using stencils
+class Stencil(NList):
+    """Cell list based neighbor list using stencils.
 
     Args:
-        r_buff (float):  Buffer width.
-        check_period (int): How often to attempt to rebuild the neighbor list.
-        d_max (float): The maximum diameter a particle will achieve, only used in conjunction with slj diameter shifting.
-        dist_check (bool): Flag to enable / disable distance checking.
-        cell_width (float): The underlying stencil bin width for the cell list
-        name (str): Optional name for this neighbor list instance.
-        deterministic (bool): When True, enable deterministic runs on the GPU by sorting the cell list.
-
-    :py:class:`stencil` creates a cell list based neighbor list object to which pair potentials can be attached for computing
-    non-bonded pairwise interactions. Cell listing allows for O(N) construction of the neighbor list. Particles are first
-    spatially sorted into cells based on the largest pairwise cutoff radius attached to this instance of the neighbor
-    list.
-
-    `M.P. Howard et al. 2016 <http://dx.doi.org/10.1016/j.cpc.2016.02.003>`_ describes this neighbor list implementation
-    in HOOMD-blue. Cite it if you utilize this neighbor list style in your work.
-
-    This neighbor-list style differs from :py:class:`Cell` based on how the adjacent cells are searched for particles. The cell
-    list *cell_width* is set by default using the shortest active cutoff radius in the system. One *stencil* is computed
-    per particle type based on the largest cutoff radius that type participates in, which defines the bins that the
-    particle must search in. Distances to the bins in the stencil are precomputed so that certain particles can be
-    quickly excluded from the neighbor list, leading to improved performance compared to :py:class:`Cell` when there is size
-    disparity in the cutoff radius. The memory demands of :py:class:`stencil` can also be lower than :py:class:`Cell` if your
-    system is large and has many small cells in it; however, :py:class:`tree` is usually a better choice for these systems.
-
-    The performance of the stencil depends strongly on the choice of *cell_width*. The best performance is obtained
-    when the cutoff radii are multiples of the *cell_width*, and when the *cell_width* covers the simulation box with
-    a roughly integer number of cells. The *cell_width* can be set manually, or be automatically scanning through a range
-    of possible bin widths using :py:meth:`tune_cell_width()`.
+        cell_width (float): The underlying stencil bin width for the cell list.
+        buffer (float): Buffer width.
+        exclusions (tuple[str]): Excludes pairs from the neighbor list, which
```

Comment:
I think this is copied from above, but if `exclusions` has the same meaning as it used to, I find "excludes pairs" to be confusing. I think you mean what styles of exclusions to use, or something like that.